provider/thp_fragment_tool.py

```diff
@@ -0,0 +1,44 @@
+"""
+thp_fragment tool.
+
+This module is meant to copy, build and execute the thp_fragment tool.
+"""
+import os
+import shutil
+
+from avocado.utils import process
+
+from virttest import data_dir
+
+
+dst_dir = "/var/tmp"
+test_bin = "/var/tmp/thp_fragment"
+source_file = "thp_fragment.c"
+
+
+def clean():
+    process.system("rm -rf %s*" % test_bin)
+
+
+def copy_tool(test):
+    host_path = os.path.join(data_dir.get_deps_dir("thp_defrag_tool"), source_file)
+    shutil.copy2(host_path, dst_dir)
+
+
+def build_tool(test):
+    build_cmd = "cd %s; gcc -lrt %s -o %s" % (dst_dir, source_file, test_bin)
+    test.log.info("Build binary file '%s'" % test_bin)
+    if process.system(build_cmd, ignore_status=True, shell=True) != 0:
+        test.fail("Failed building the the tool binary: %s" % test_bin)
+
+
+def get_tool_output():
+    return process.getoutput(test_bin, shell=True)
```

Comment:
Hi @mcasquer 
Do we not care about the status of the running `test_bin`? 
Since the default of ignore_status is `True`, refer to https://github.com/YongxueHong/avocado/blob/bd72146c1f692ab89639961be41d310202560f28/avocado/utils/process.py#L1157
Here, I am afraid that we can not get the accurate value if failed to run `thp_fragment`.
so I suggest getting the value before making sure it can run well.
Please let me your opinions. Thanks.

Reply:
Included !